### PR TITLE
Updated the Android build images.

### PR DESCRIPTION
**Pull request by FenrirWillow**
The build images are now separated per SDK version and also contain 'slim' versions without the NDK inside them. This significantly improves their size and build times. Also all tools are not at their latest versions including the way we manage the SDKs and packages within the Android ecosystem

### CODE REVIEW

**Comment by FenrirWillow** [comment]
I don't know what the best way to handle the latest tag is. I think we should just not have it, since it's not really something incremental. Would appreciate some input on it.

**Comment by dannytip** [comment]
You should probably tag it as per: https://github.com/phusion/passenger-docker

Actually have separate repo's for the different versions.  Since they are public repos it shouldn't be a problem.

I.e.

phusion/passenger-ruby20:<VERSION>
phusion/passenger-ruby21:<VERSION>
phusion/passenger-ruby22:<VERSION>

File: android-build/sdk-23/Dockerfile [hunk 1]
```diff
@@ -1,95 +1,96 @@
...
-
-CMD ls
+FROM ubuntu:16.04
+
+# --- Environment variables ---
+ENV ANDROID_HOME /opt/android-sdk-linux
+
+# --- Run system tool updates ---
+RUN apt-get update
+RUN apt-get -y install unzip
```
Comment (dannytip) [inline android-build/sdk-23/Dockerfile:8]: Any reason to not just bundle all these into 1 line?

**Comment by FenrirWillow** [comment]
About the tagging and the different versions - I prefer keeping all versions in the same repository, this way we can build them with a simple bash script.

The script also does the tagging for us. Currently it will tag the images like this: `sdk-<version>` or `sdk-<version>-ndk` for the images that have the ndk inside them.

**Comment by dannytip** [comment]
I'm sure you could still build them with a simple bash script if they were in different repos (docker repo's that is, the source could all stay in one).

@pmead what do you think?

**Pull request merged**